info/container.go

```diff
@@ -405,3 +405,34 @@ func calculateCpuUsage(prev, cur uint64) uint64 {
 	}
 	return cur - prev
 }
+
+type Percentiles struct {
+	// Average over the collected sample.
+	Mean uint64 `json:"mean"`
+	// Max seen over the collected sample.
+	Max uint64 `json:"max"`
+	// 90th percentile over the collected sample.
+	Ninety uint64 `json:"ninety"`
+}
+
+type Usage struct {
+	// Indicates amount of data available [0-100].
+	// If we have data for half a day, we'll still process DayUsage,
+	// but set PercentComplete to 50.
+	PercentComplete int32 `json:"percent_complete"`
+	// Mean, Max, and 90p cpu rate value in milliCpus/seconds. Converted to milliCpus to avoid floats.
+	CpuUsage Percentiles `json:"cpu_usage"`
```

Comment:
We need something like 'NaN'. If we use float we can do that.

On Mon, Feb 2, 2015 at 9:45 AM, Victor Marmol <notifications@github.com>
wrote:

> In info/container.go
> <https://github.com/google/cadvisor/pull/481#discussion_r23942180>:
>
> > +type Percentiles struct {
> > +	// Average over the collected sample.
> > +	Mean uint64 `json:"mean"`
> > +	// Max seen over the collected sample.
> > +	Max uint64 `json:"max"`
> > +	// 90th percentile over the collected sample.
> > +	Ninety uint64 `json:"ninety"`
> > +}
> > +
> > +type Usage struct {
> > +	// Indicates amount of data available [0-100].
> > +	// If we have data for half a day, we'll still process DayUsage,
> > +	// but set PercentComplete to 50.
> > +	PercentComplete int32 `json:"percent_complete"`
> > +	// Mean, Max, and 90p cpu rate value in milliCpus/seconds. Converted to milliCpus to avoid floats.
> > +	CpuUsage Percentiles `json:"cpu_usage"`
>
> I buy that, we don't have an "is present" in JSON. +1
>
> (well theres the omitifempty but that doesn't work anywhere close so what
> we'd like it to work like)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/481/files#r23942180>.
>